catalog/core/catalog-core-api/src/main/java/ddf/catalog/plugin/AccessPlugin.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.plugin;
+
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.UpdateRequest;
+
+/**
+ * An AccessPlugin allows or denies access to the Catalog operation or response.
```

Comment:
Regarding my second question, say you have two plugins, `a` and `b`, and a query request comes in. Should we do:

```
a.processPreQuery();
b.processPreQuery();
// run query
a.processPostQuery();
b.processPostQuery();
```

or

```
a.processPreQuery();
b.processPreQuery();
// run query
b.processPostQuery();
a.processPostQuery();
```
?

The second option is usually the typical one but I guess we should just be consistent with how we handle this kind of scenario elsewhere.